packages/core/src/client.ts

```diff
@@ -236,113 +260,123 @@ export class SkipRouter {
+          );
+        }
+
+        const signer = await getOfflineSigner(multiChainMsg.chainID);
+
+        const tx = await this.executeMultiChainMessage({
+          signerAddress: userAddresses[multiChainMsg.chainID],
+          signer,
+          message: multiChainMsg,
+          feeAmount: coin(feeAmount, feeInfo.denom),
+        });
+
+        if (options.onTransactionBroadcast) {
+          await options.onTransactionBroadcast({
+            chainID: multiChainMsg.chainID,
+            txHash: tx.transactionHash,
+          });
+        }
+
+        const txStatusResponse = await this.waitForTransaction({
+          chainID: multiChainMsg.chainID,
+          txHash: tx.transactionHash,
+        });
+
+        if (options.onTransactionSuccess) {
+          await options.onTransactionSuccess({
+            chainID: multiChainMsg.chainID,
+            txHash: tx.transactionHash,
+            success: txStatusResponse.status === "STATE_COMPLETED_SUCCESS",
+          });
+        }
       }
 
-      if (!getOfflineSigner) {
-        throw new Error(
-          `Unable to find offline signer for chain ${multiHopMsg.chainID}`,
-        );
-      }
-
-      const signer = await getOfflineSigner(multiHopMsg.chainID);
-
-      const tx = await this.executeMultiChainMessage({
-        signerAddress: userAddresses[multiHopMsg.chainID],
-        signer,
-        message: multiHopMsg,
-        feeAmount: coin(feeAmount, feeInfo.denom),
-      });
-
-      const txStatusResponse = await this.waitForTransaction({
-        chainID: multiHopMsg.chainID,
-        txHash: tx.transactionHash,
-      });
-
-      if (options.onTransactionSuccess) {
-        await options.onTransactionSuccess(
-          tx.transactionHash,
-          multiHopMsg.chainID,
-          txStatusResponse,
-        );
+      if ("evmTx" in message) {
+        const { evmTx } = message;
+
+        let getEVMSigner = this.getEVMSigner;
+        if (options.getEVMSigner) {
+          getEVMSigner = options.getEVMSigner;
+        }
+        if (!getEVMSigner) {
+          throw new Error("Unable to get EVM signer");
+        }
+
+        const evmSigner = await getEVMSigner(evmTx.chainID);
+
+        const txReceipt = await this.executeEVMTransaction({
+          message: evmTx,
+          signer: evmSigner,
+        });
+
+        if (options.onTransactionBroadcast) {
+          await options.onTransactionBroadcast({
+            chainID: evmTx.chainID,
+            txHash: txReceipt.transactionHash,
+          });
+        }
+
+        const txStatusResponse = await this.waitForTransaction({
+          chainID: evmTx.chainID,
+          txHash: txReceipt.transactionHash,
+        });
+
+        if (options.onTransactionSuccess) {
+          await options.onTransactionSuccess({
+            chainID: evmTx.chainID,
+            txHash: txReceipt.transactionHash,
+            success: txStatusResponse.status === "STATE_COMPLETED_SUCCESS",
```

Comment:
This should be `state` not `status`. See api spec: https://api-docs.skip.money/reference/gettransactionstatus

```suggestion
            success: txStatusResponse.state === "STATE_COMPLETED_SUCCESS",
```

Also: The previous implementation where you return the TxStatusResponse object is better imo. Then the user of the sdk can decide how to handle different states.

Reply:
Hey @MbBrainz I appreciate the extra eyes on this. This version of the library uses the v2 version of lifecycle tracking where `state` is part of the response. https://api-docs.skip.money/reference/gettransactionstatusv2

I do agree on the feedback about returning `TxStatusResponse` though and will incorporate that before this is officially released.

Reply:
I dont think i completely understand what you mean. I checked out the gettransactionstatusv2 reference, but the TxStatusResponse param is in both cases called `state` instead of `status` right, which is the issue i mentioned above. Maybe we are talking about different things and sorry if i missunderstand 😅

Reply:
Ah, the docs I linked you to are incorrect I probably should have read them first 😂 I will make sure that gets updated but yes v2 of the lifecycle tracking API returns a field named `status`